torch/_tensor.py

```diff
@@ -941,14 +941,9 @@ def to_sparse_csr(self):
         if self.is_sparse:
             coalesced_self = self.coalesce()
             row_indices = coalesced_self.indices()[0]
-            ro = [0]
-            i = 0
-            for irow in range(self.shape[0]):
-                while i < row_indices.size()[0] and row_indices[i] == irow:
-                    i += 1
-                ro.append(i)
             device = coalesced_self.values().device
-            crow_indices = torch.tensor(ro, dtype=row_indices.dtype, device=device)
+            arange = torch.arange(self.shape[0] + 1, device=device)
```

Comment:
Use
```
arange = torch.arange(self.shape[0] + 1, device=device, dtype=row_indices.dtype)
```
that will be slightly more efficient for int32 dtype case:
```
In [11]: N = 1000

In [12]: x, y = torch.arange(N), torch.arange(N, dtype=torch.int32)

In [13]: %timeit torch.bucketize(x, y, out_int32=True)
10.5 µs ± 61.4 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

In [14]: x, y = torch.arange(N, dtype=torch.int32), torch.arange(N, dtype=torch.int32)

In [15]: %timeit torch.bucketize(x, y, out_int32=True)
8.24 µs ± 9.79 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```

Reply:
Already did this :)